pyirf/scripts/make_DL3.py

```diff
@@ -0,0 +1,338 @@
+
+    # Later we should move this out of here, perhaps under a "utils" module.
+
+    if args.pipeline == "EventDisplay":
+
+        # THETA IS SUPPOSED TO BE IN DL2 DATA ACCORDING TO GADF
+        # EventDisplay provides true and reconstructed directions, so we
+        # calculate it here and we add it to the tables.
+
+        # WARNING:
+        # this is true only for point-source simulations!
+        # in the case of diffuse simulations we must distinguish between
+        # - distance between reconstructed direction and center of the FoV
+        # - distance between reconstructed and true direction
+
+        for particle in particles:
+
+            THETA = angular_separation(
+                evt_dict[particle]["TRUE_AZ"],
+                evt_dict[particle]["TRUE_ALT"],
+                evt_dict[particle]["AZ"],
+                evt_dict[particle]["ALT"],
+            )  # in degrees
+
+            # Add THETA column
+            evt_dict[particle]["THETA"] = THETA
+
+            # print(evt_dict[particle].head(n=5))
+
+    # =========================================================================
+    #                   REST OF THE OPERATIONS (TO BE REFACTORED)
+    # =========================================================================
+
+    # Apply offset cut to proton and electron
+    for particle in ["electron", "proton"]:
+
+        # There seems to be a problem in using pandas from FITS data
+        # ValueError: Big-endian buffer not supported on little-endian compiler
+
+        # I convert to astropy table....
+        # should we use only those?
+
+        evt_dict[particle] = Table.from_pandas(evt_dict[particle])
+
+        if args.debug:
+            print(particle)
+            # print(evt_dict[particle].head(n=5))
+            print(evt_dict[particle])
+
+        # print('Initial stat: {} {}'.format(len(evt_dict[particle]), particle))
+
+        mask_theta = (
+            evt_dict[particle]["THETA"]
+            < cfg["particle_information"][particle]["offset_cut"]
+        )
+        evt_dict[particle] = evt_dict[particle][mask_theta]
+        # PANDAS EQUIVALENT
+        # evt_dict[particle] = evt_dict[particle].query(
+        #     "THETA <= {}".format(cfg["particle_information"][particle]["offset_cut"])
+        # )
+
+    # Add required data in configuration file for future computation
+    for particle in particles:
+        n_files = cfg["particle_information"][particle]["n_files"]
+        print(f"{n_files} files for {particle}")
+        cfg["particle_information"][particle]["n_files"] = len(
+            np.unique(evt_dict[particle]["OBS_ID"])
+        )
+        cfg["particle_information"][particle]["n_simulated"] = (
+            cfg["particle_information"][particle]["n_files"]
+            * cfg["particle_information"][particle]["n_events_per_file"]
+        )
+
+    # Define model for the particles
+    model_dict = {
+        "gamma": CrabSpectrum("hegra").model,
```

Comment:
This should be configurable. I think also the MAGIC sprectum was used for CTA and especially in the LST region, the difference is quite large.

Reply:
also here, better open an issue to not forget